docs/module.md

```diff
@@ -1,341 +1,246 @@
-choose to assign it the type `EstablishmentType`. Otherwise, `School` exposes
-every field). This effectively makes the `person1` field an enforced
-implementation detail! Outsiders can't access it, since it's not present in the
-signature; the signature **constrained** what others can access.
-
-The type `EstablishmentType.profession` is **abstract**: it doesn't have a
-concrete type; it's saying "I don't care what the actual type is, but it's used
-as input to `getProfession`". This is useful to fit many modules under the same
-interface:
+This type can be applied with `:` when a module is created. The module type
+specifies exactly what is exposed from a module, anything extra will be hidden.
 
 ```reason
-module Company: EstablishmentType = {
-  type profession = CEO | Designer | Engineer | ...;
+module Duration: Duration = {
+  type t = int;
+  let fromSeconds = value => value;
+  let add = (x, y) => x + y;
 
-  let getProfession = (person) => ...;
-  let person1 = ...;
-  let person2 = ...;
+  /* The module type hides this */
+  let extraStuff = () => ();
 };
-```
 
-It's also useful to hide the underlying type as an implementation detail others
-can't rely on. If you ask what the type of `Company.profession` is, instead of
-exposing the variant, it'll only tell you "it's `Company.profession`".
+/* Error: Unbound value */
+Duration.extraStuff();
+```
 
-### Extending module signatures
+### Including Module Types
 
-Like modules themselves, module signatures can also be extended by other module signatures using `include`:
+Module types can also be extended using the `include` keyword:
 
 ```reason
-module type BaseComponent = {
-  let defaultGreeting: string;
-  let getAudience: (~excited: bool) => string;
+module type Printable = {
+  type printable;
+  let print: printable => string;
 };
 
-module type ActualComponent = {
-  /* the BaseComponent signature is copied over */
-  include BaseComponent;
-  let render: unit => string;
+module type PrintableDuration = {
+  include Duration;
+  include Printable;
 };
 ```
 
-**Note**: `BaseComponent` is a module **type**, not an actual module itself!
+### Accessing Implicit Module Type
 
-If you do not have a defined module type, you can extract it from an actual module
-using `include (module type of ActualModuleName)`. For example, we can extend the
-`List` module from the OCaml standard library, which does not define a module
+Sometimes you may need access to the type of a module, but that module will not
+have an explicit type defined. Use `(module type of ModuleName)` to access the
 type.
 
 ```reason
-module type MyList = {
-  include (module type of List);
-  let myListFun: list('a) => list('a);
+module type PrintableDuration = {
+  include (module type of Duration);
+  include Printable;
 };
 ```
 
-### Every `.rei` file is a signature
+## Interface Files
+
+Interface files (`.rei` files) are similar to module types. They create a module
```

Comment:
Might even make a stronger claim - that `.rei` files _are_ module types applied to an entire file and everything you learn/know about module types can directly be used inside of a `.rei` file.